tensorboard/webapp/core/views/page_title_container.ts

```diff
@@ -36,12 +58,44 @@ import {State} from '../state';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class PageTitleContainer {
-  readonly title$ = this.store.pipe(
-    select(getEnvironment),
-    map((env) => env.window_title || 'TensorBoard'),
-    // (it's an empty string when the `--window_title` flag is not set)
+  private readonly getExperimentId$ = this.store.select(getRouteKind).pipe(
+    filter((routeKind) => routeKind === RouteKind.EXPERIMENT),
+    withLatestFrom(this.store.select(getExperimentIdsFromRoute)),
+    map(([, experimentIds]) =>
+      experimentIds && experimentIds.length === 1 ? experimentIds[0] : null
+    )
+  );
+
+  private readonly experiment$ = this.getExperimentId$.pipe(
+    filter(Boolean),
+    mergeMap((experimentId) => {
+      return this.store.select(getExperiment, {experimentId});
+    })
+  );
+
+  private readonly experimentName$ = this.experiment$.pipe(
```

Comment:
optional: I would slightly prefer inlining this `map` into the earlier chain, and rename `experiment$` to `experimentName$`, since it is only used in 1 place.

Reply:
Merged into one.